invenio_vocabularies/contrib/names/datastreams.py

```diff
@@ -8,13 +8,43 @@
 
 """Names datastreams, transformers, writers and readers."""
 
+import requests
 from invenio_access.permissions import system_identity
 from invenio_records.dictutils import dict_lookup
 
 from ...datastreams.errors import TransformerError
+from ...datastreams.readers import BaseReader
 from ...datastreams.transformers import XMLTransformer
 
 
+class OrcidHTTPReader(BaseReader):
+    """ORCiD HTTP Reader."""
+
+    def __init__(self, orcid_id, test_mode=True, *args, **kwargs):
+        """Constructor."""
+        if test_mode:
+            origin = f"https://sandbox.orcid.org/{orcid_id}"
+        else:
+            origin = f"https://orcid.org/{orcid_id}"
+
+        super().__init__(origin, *args, **kwargs)
+
+    def read(self):
```

Comment:
was doubting if `orcid_id` should be a parameter here instead of the constructor. But then we would create a reading loop with diff parameters each time, which breaks the API of the other readers.

i.e. I don't like the fact of creating an object per ORCiD to fetch, but it is what's meant for.

Reply:
base signature accepts args and kwargs, could have gone there...